### PR TITLE
Add spec `Factory`.

### PR DESCRIPTION
**Pull request by abrightwell**
Here we're simply adding a Factory for generating common objects that are the result of client requests, etc for testing purposes.

### CODE REVIEW

File: spec/factory.cr [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      major_version: 12,
+      plan_id:       "memory-4",
+      cpu:           4,
+      memory:        111,
+      oldest_backup: nil,
+      provider_id:   "aws",
+      region_id:     "us-east-2",
+      network_id:    "nfpvoqooxzdrriu6w3bhqo55c4",
+      storage:       1234,
+    }.merge(params)
```
Comment (abrightwell) [inline spec/factory.cr:22]: I figured this approach was the easiest way for use to have some 'defaults' such that in the majority case we can just call `Factory.cluster`... however, if we needed multiple with different values for whatever reason, those could be overridden via kwarg, so `Factory.cluster id: abc123, name: foobar` but all well might be the same.

**Review by gitstashpop — APPROVED** [review]
lgtm. Would TestFactory be a more appropriate module name?

**Comment by abrightwell** [comment]
> lgtm. Would TestFactory be a more appropriate module name?

Maybe? I'm not sure... I think given it's part of the `spec` package that it's implied. :man_shrugging: I'm not sure I have any strong opinions on it.

**Review by will — APPROVED** [review]

**Comment by abrightwell** [comment]
Thanks!

**Pull request merged**